### PR TITLE
Added Node.js server for collecting playtest logs

### PR DESCRIPTION
**Pull request by rcgray**
This branch introduces a node.js server backend that can collect debug logs periodically over the course of a playtest.

### CODE REVIEW

File: server/routes.js [hunk 1]
```diff
@@ -17,6 +17,11 @@ const routes = function(app) {
 		res.sendFile(path.join(__dirname, '..', 'built', 'shrdlu.html'));
 	});
 
+	// environment variables set in .env that the frontend might need
```
Comment (rcgray) [inline server/routes.js]: This approach with env.js and use of the BASE_URL variable attached to the global was meant to allow us to include variables from the .env file in the frontend.  We ended up going a different route and this is removed in a future checkin

File: src/a4engine3/A4EngineApp.ts [hunk 2]
```diff
@@ -50,6 +50,8 @@ var QUIT_REQUEST_ACTION_LOAD2:number = 3;
 var QUIT_REQUEST_ACTION_LOAD3:number = 4;
 var QUIT_REQUEST_ACTION_LOAD4:number = 5;
 
+// constants from our .env file
```
Comment (rcgray) [inline src/a4engine3/A4EngineApp.ts]: removed in the next checkin, we ended up not going this way

**Comment by santiontanon** [comment]
Hi Gray, some questions before I accept:
- I see modifications to src/lib/lz-string.ts. Why was that needed?
- I think some files of your .idea project folder got pushed by mistake, right?
- What was package.json? Do I really need to keep it in the root folder? or is this something that was only needed by the server and I can hide it inside of the server folder?

**Comment by rcgray** [comment]
The files weren’t checked in as a mistake at the time, but maybe they are now.  They are leftovers from some grander plans as I was working on the server, but I don’t know now if they would fit well.  I will remove the files.  Lz-string.ts does not compile on the latest TSC with your parameters, or maybe I have some global settings somewhere that’s checked by default… but it's due to a declaration of a variable that’s never used and was a quick fix.  But I can remove that change as well.

**Comment by rcgray** [comment]
Confirmed the lz-string.ts issue.  It's due to --noUnusedLocals in the build script.  Results in failed build with "error TS6133: 'next' is declared but its value is never read." using TSC v3.7.3

**Comment by santiontanon** [comment]
Interesting, I have an older version of TSC (2.9.2, maybe I should update, haha). I am also using that flag (and it complains when I do not use a variable), but for some reason it does not complain about that one in particular... oh well! Ok, I'll go through it one more time tomorrow morning just in case and accept :)

**Comment by santiontanon** [comment]
Just accepted it, thanks Gray! I ended up creating a new source file, and moving all the"saving logs to server" code there, so the game and app classes stay clean, and added a boolean variable to activate/deactivate logging (that error on startup when the server is not running really bothered me :)). I now need to find some time to try it out on my server :)

**Pull request merged**